pkg/nodeaddress/node_address.go

```diff
@@ -15,180 +15,189 @@
+// default values derived from the system. device can be set to the primary
+// network device of the system in which case the first address with global
+// scope will be regarded as the system's node address.
+func InitDefaultPrefix(device string) {
+	ip, err := firstGlobalV4Addr(device)
+	if err == nil {
+		ipv4Address = ip
+
+		v4range := fmt.Sprintf(DefaultIPv4Prefix+"/%d",
+			ip.To4()[3], DefaultIPv4PrefixLen)
+		_, ip4net, err := net.ParseCIDR(v4range)
+		if err != nil {
+			log.Panicf("BUG: Invalid default prefix '%s': %s", v4range, err)
+		}
 
-	return &net.IPNet{
-		IP:   IPv4Address.IP().Mask(mask),
-		Mask: mask,
-	}
-}
+		ipv4AllocRange = ip4net
 
-// IPv4AllocRange returns the IPv4 allocation prefix of this node
-func IPv4AllocRange() *net.IPNet {
-	mask := net.CIDRMask(DefaultIPv4PrefixLen, 32)
+		v6range := fmt.Sprintf("%s%02x%02x:%02x%02x:0:0/%d",
+			DefaultIPv6Prefix, ip[0], ip[1], ip[2], ip[3],
+			DefaultIPv6PrefixLen)
 
-	return &net.IPNet{
-		IP:   IPv4Address.IP().Mask(mask),
-		Mask: mask,
+		ip6, ip6net, err := net.ParseCIDR(v6range)
+		if err != nil {
+			log.Panicf("BUG: Invalid default prefix '%s': %s", v6range, err)
+		}
+
+		ip6[14] = 0xff
+		ip6[15] = 0xff
+
+		ipv6Address = ip6
+		ipv6AllocRange = ip6net
 	}
 }
 
-// IPv6ClusterRange returns the IPv6 prefix of the clustr
-func IPv6ClusterRange() *net.IPNet {
-	mask := net.CIDRMask(DefaultIPv6ClusterPrefixLen, 128)
+func init() {
+	InitDefaultPrefix("")
+}
 
+// GetIPv4ClusterRange returns the IPv4 prefix of the cluster
+func GetIPv4ClusterRange() *net.IPNet {
+	mask := net.CIDRMask(DefaultIPv4ClusterPrefixLen, 32)
 	return &net.IPNet{
-		IP:   IPv6Address.IP().Mask(mask),
+		IP:   ipv4AllocRange.IP.Mask(mask),
 		Mask: mask,
 	}
 }
 
-// IPv6AllocRange returns the IPv6 allocation prefix of this node
-func IPv6AllocRange() *net.IPNet {
-	mask := net.CIDRMask(DefaultIPv6PrefixLen, 128)
+// GetIPv4AllocRange returns the IPv4 allocation prefix of this node
+func GetIPv4AllocRange() *net.IPNet {
+	return ipv4AllocRange
+}
 
+// GetIPv6ClusterRange returns the IPv6 prefix of the clustr
+func GetIPv6ClusterRange() *net.IPNet {
+	mask := net.CIDRMask(DefaultIPv6ClusterPrefixLen, 128)
 	return &net.IPNet{
-		IP:   IPv6Address.IP().Mask(mask),
+		IP:   ipv6AllocRange.IP.Mask(mask),
 		Mask: mask,
 	}
 }
 
-func firstGlobalV4Addr(intf string) (net.IP, error) {
-	var link netlink.Link
-	var err error
+// GetIPv6AllocRange returns the IPv6 allocation prefix of this node
+func GetIPv6AllocRange() *net.IPNet {
+	return ipv6AllocRange
+}
 
-	if intf != "" && intf != "undefined" {
-		link, err = netlink.LinkByName(intf)
-		if err != nil {
-			return firstGlobalV4Addr("")
-		}
-	}
+// SetIPv4 sets the IPv4 address of the node
+func SetIPv4(ip net.IP) {
+	ipv4Address = ip
+}
 
-	addr, err := netlink.AddrList(link, netlink.FAMILY_V4)
-	if err != nil {
-		return nil, err
-	}
+// GetIPv4 returns the IPv4 address of the node
+func GetIPv4() net.IP {
+	return ipv4Address
```

Comment:
I had direct access at first but then figured that having Get and Set function allows to do validation as is now done for some and it allows to lock the variables if needed later on.

I didn't really see a downside of using Getters and Setters here.